src/WebSocket.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace ApiClients\Client\Pusher;
+
+use React\Dns\Resolver\Resolver;
+use React\EventLoop\LoopInterface;
+use Rx\DisposableInterface;
+use Rx\ObserverInterface;
+use Rx\Subject\ReplaySubject;
+use Rx\Subject\Subject;
+use Rx\Websocket\Client;
+
+/**
+ * Class WebSocket - WebSocket wrapper that queues messages while the connection is being established.
+ */
+final class WebSocket extends Subject
+{
+    private $ws;
+    private $sendSubject;
+
+    public function __construct(string $url, bool $useMessageObject = false, array $subProtocols = [], LoopInterface $loop = null, Resolver $dnsResolver = null)
```

Comment:
Why not accept a `Client` instance here instead of its constructor parameters and constructing it inside the constructor?

Reply:
PRs to my PR are welcome